src/main/java/dev/kurumiDisciples/chisataki/modmail/Ticket.java

```diff
@@ -117,18 +175,52 @@ public boolean isClosed(){
     return getStatus() == StatusType.CLOSED;
   }
 
+  @Deprecated
   private static int countFiles() {
-    String directoryPath = "data/tickets";
-    File directory = new File(directoryPath);
-    if (!directory.exists()) {
-      System.out.println("Directory does not exist.");
-      return 0;
-    }
-    if (!directory.isDirectory()) {
-      System.out.println("Not a directory.");
-      return 0;
-    }
-    File[] files = directory.listFiles();
-    return files.length;
+    return TicketDatabaseUtils.countTickets();
+  }
+
+  @Override
+  public String getTableName() {
+    return "tickets";
+  }
+
+  @Override
+  public String getTableSchema() {
+    return "CREATE TABLE IF NOT EXISTS tickets ("
+        + "ticket INT NOT NULL,"
+        + "ticket_id BIGINT NOT NULL,"
+        + "member_id BIGINT NOT NULL,"
+        + "staff_id BIGINT,"
+        + "subject VARCHAR(255) NOT NULL,"
+        + "body VARCHAR(255) NOT NULL,"
+        + "status VARCHAR(255) NOT NULL,"
+        + "reason VARCHAR(255),"
+        + "PRIMARY KEY (ticket)"
+        + ");";
+  }
+
+  @Override
+  public String getPrimaryKey() {
+    return "ticket";
+  }
+
+  @Override
+  public Integer getPrimaryKeyType() {
+    return java.sql.Types.INTEGER;
+  }
```

Comment:
Wouldn't it be best for ticket_id to be the id or what is its purpose then?

Reply:
Ticket id is the channel id of the channel created. We should probably change the variable name